src/PhpSpec/ObjectBehavior.php

```diff
@@ -31,9 +31,90 @@
  * @method void beConstructedWith(...$arguments)
  * @method void beConstructedThrough($factoryMethod, array $constructorArguments = array())
  * @method void beAnInstanceOf($class)
+ *
  * @method void shouldHaveType($type)
+ * @method void shouldNotHaveType($type)
+ * @method void shouldReturnAnInstanceOf($type)
```

Comment:
@deeky666 but most builtin matchers make no sense on the ObjectBehavior, as they are about matching expectations on method return values, not on the SUT instance itself. Most of the things you are adding make no sense here.

On ObjectBehavior, we should add only these ones (on top of the existing ones):

- ``shouldNotHaveType``
- ``shouldBeAnInstanceOf``
- ``shouldNotBeAnInstanceOf``
- ``shouldNotImplement``
- ``shouldNotThrow``

and maybe these ones to account for Countable or ArrayAccess SUT:
- ``shouldHaveCount``
- ``shouldNotHaveCount``
- ``shouldHaveKeyWithValue``
- ``shouldNotHaveKeyWithValue``
- ``shouldHaveKey``
- ``shouldNotHaveKey``

Everything else should be removed.